### PR TITLE
Migrate filterprocessor from Collector core to Collector contrib

### PR DESCRIPTION
**Pull request by Saber-W**
**Description:**
This PR migrates the filterprocessor component from the Collector core repository to the Collector contrib repository.

This PR retains the commit history from core and includes plumbing fixes for go.mod, makefile, components.go.

**Link to tracking Issue:**
[opentelemetry-collector/#3474](https://github.com/open-telemetry/opentelemetry-collector/issues/3474)
[opentelemetry-collector/#3644](https://github.com/open-telemetry/opentelemetry-collector/issues/3644)

### CODE REVIEW

**Comment by alolita** [comment]
Hi @bogdandrutu @tigrannajaryan based on our discussion last week, we're starting to file PRs to move each core component to contrib. Please review and merge. Thanks!

File: cmd/otelcontribcol/components.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/k8sobserver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage/filestorage"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/cumulativetodeltaprocessor"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor"
```
Comment (mx-psi) [inline cmd/otelcontribcol/components.go:54]: You need to add a `replace` directive on the main `go.mod` file for this to work.

File: processor/filterprocessor/config.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterprocessor
+
+import (
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/internal/processor/filtermetric"
```
Comment (mx-psi) [inline processor/filterprocessor/config.go:19]: This is an internal package of a different repo so we need to move it here (or find an alternative solution)

**Review by mx-psi — DISMISSED** [review]

File: go.mod [hunk 3]
```diff
@@ -40,9 +40,9 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/oidcauthextension v0.0.0-00010101000000-000000000000
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage v0.0.0-00010101000000-000000000000
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/metrics v0.30.0 // indirect
-	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.0.0-00010101000000-000000000000
```
[the remark this answers is not in the document]
Comment (Saber-W) [reply go.mod:43]: This is deleted by go mod tidy. I can add it if you think this is necessary.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**